src/AltinnCore/Designer/Controllers/ApplicationMetadataController.cs

```diff
@@ -0,0 +1,70 @@
+using Altinn.Platform.Storage.Models;
+using AltinnCore.Common.Services.Interfaces;
+using Microsoft.AspNetCore.Authorization;
+using Microsoft.AspNetCore.Mvc;
+using Microsoft.Extensions.Logging;
+
+namespace Designer.Controllers
+{
+    /// <summary>
+    /// Controller for serving/editing the application metadata json file
+    /// </summary>
+    [Authorize]
+    [Route("/designer/api/v1/{org}/{app}")]
+    public class ApplicationMetadataController : ControllerBase
```

Comment:
Consider annotate with: [ApiController] and use Controller class as base class.

Reply:
Agree on the ApiController attribute, but keeping ControllerBase, ref microsoft docs:

> Don't create a web API controller by deriving from the Controller class. Controller derives from ControllerBase and adds support for views, so it's for handling web pages, not web API requests. There's an exception to this rule: if you plan to use the same controller for both views and APIs, derive it from Controller.